### PR TITLE
Select home & squadron systems from Combo box

### PR DESCRIPTION
**Pull request by Hoodathunk**
* Added functionality to select home and squadron systems in `Commander Details` tab using a dynamic Combo/Text box.
* Fixed minor grammatical error in the `Mission check system` script.

### CODE REVIEW

File: StarMapService/EdsmSystemData.cs [hunk 1]
```diff
@@ -69,6 +69,38 @@ public static List<StarSystem> GetStarMapSystems(string[] systems, bool showCoor
             return null;
         }
 
+        /// <summary> Partial of system name is required. </summary>
+        public static List<StarSystem> GetStarMapSystemsPartial(string system, bool showCoordinates = true, bool showSystemInformation = true)
+        {
+            if (system == null) { return null; }
+            var client = new RestClient(baseUrl);
+            var request = new RestRequest("api-v1/systems", Method.POST);
+            request.AddParameter("systemName", system + "%");
```
Comment (Tkael) [inline StarMapService/EdsmSystemData.cs]: Is "%" necessary? The API doesn't document that as a requirement for the `https://www.edsm.net/api-v1/systems` endpoint (ref. https://www.edsm.net/en/api-v1)

File: EDDI/MainWindow.xaml.cs [hunk 2]
```diff
@@ -460,44 +456,70 @@ private void windowLoaded(object sender, RoutedEventArgs e)
...
             {
-                eddiConfiguration.HomeSystem = string.IsNullOrWhiteSpace(eddiHomeSystemText.Text) ? null : eddiHomeSystemText.Text.Trim();
-                eddiConfiguration = EDDI.Instance.updateHomeSystem(eddiConfiguration);
+                string systemName = homeSystemDropDown.Text?.ToLowerInvariant();
+                if (systemName.Length > 1)
+                {
+                    systemList = systemList.Where(s => s.StartsWith(systemName, StringComparison.InvariantCultureIgnoreCase)).ToList();
+                    if (systemList.Count < 5)
+                    {
+                        systemList = StarMapService.GetStarMapSystemsPartial(systemName)
```
Comment (Tkael) [inline EDDI/MainWindow.xaml.cs]: Recommended: `systemList = StarMapService.GetStarMapSystemsPartial(systemName, false, false)`
There's no need to request more information than we need, and all we really need here is names.

File: EDDI/MainWindow.xaml.cs [hunk 3]
```diff
@@ -630,13 +652,40 @@ private void squadronRankDropDownUpdated(object sender, SelectionChangedEventArg
...
             {
-                eddiConfiguration.SquadronSystem = string.IsNullOrWhiteSpace(eddiSquadronSystemText.Text) ? null : eddiSquadronSystemText.Text.Trim();
-                eddiConfiguration = EDDI.Instance.updateSquadronSystem(eddiConfiguration);
+                string systemName = squadronSystemDropDown.Text?.ToLowerInvariant();
+                if (systemName.Length > 1)
+                {
+                    systemList = systemList.Where(s => s.StartsWith(systemName, StringComparison.InvariantCultureIgnoreCase)).ToList();
+                    if (systemList.Count < 5)
+                    {
+                        systemList = StarMapService.GetStarMapSystemsPartial(systemName)
```
Comment (Tkael) [inline EDDI/MainWindow.xaml.cs]: Recommended: `systemList = StarMapService.GetStarMapSystemsPartial(systemName, false, false)`
There's no need to request more information than we need, and all we really need here is names.

File: StarMapService/EdsmSystemData.cs (hunk 1 above)
Comment (Tkael) [reply StarMapService/EdsmSystemData.cs]: It looks like both work, but it is unnecessary. Compare:
https://www.edsm.net/api-v1/systems?systemName=Ach
https://www.edsm.net/api-v1/systems?systemName=Ach%

**Comment by Tkael** [comment]
For consideration for the `Lostfocus` event...
```
        private void HomeSystemDropDown_LostFocus(object sender, RoutedEventArgs e)
        {
            if (sender is System.Windows.Controls.ComboBox comboBox)
            {
                // Lost focus triggers both when clicking in to the combo box and when clicking away
                if (!comboBox.IsKeyboardFocusWithin)
                {
                    // We are clicking away from the combobox
                    if (systemList.Contains(comboBox.Text))
                    {
                        // Update configuration to new home system
                        EDDIConfiguration eddiConfiguration = EDDIConfiguration.FromFile();
                        eddiConfiguration.HomeSystem = comboBox.Text;
                        eddiConfiguration = EDDI.Instance.updateHomeSystem(eddiConfiguration);
                        eddiConfiguration.ToFile();

                        // Update station options for new system
                        ConfigureHomeStationOptions(eddiConfiguration.HomeSystem);
                    }
                    systemList.Clear();
                }
            }
        }
```

File: StarMapService/EdsmSystemData.cs (hunk 1 above)
Comment (Hoodathunk) [reply StarMapService/EdsmSystemData.cs]: It's absolutely necessary, per Anthor.

**Comment by Hoodathunk** [comment]
Withdrawn.

File: StarMapService/EdsmSystemData.cs (hunk 1 above)
Comment (Tkael) [reply StarMapService/EdsmSystemData.cs]: After further discussion with @Hoodathunk, it appears to be necessary to add to all 2 letter searches (at minimum). This is an undocumented API requirement that we should note in the code.

**Comment by richardbuckle** [comment]
Fixes #1241

**Pull request merged**